src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -904,12 +903,14 @@ export class Kernel implements IKernel {
                 launchingFile
             );
             if (suggestedDir && (await this.fs.localDirectoryExists(suggestedDir))) {
+                traceInfo('UpdateWorkingDirectoryAndPath in Kernel');
```

Comment:
Moved these closer to when the paths are changed, this way when its logged we know its changing 
& if its not logged, then cwd is not changing